formats/objfile/reader.go

```diff
@@ -121,3 +121,12 @@ func (r *Reader) Close() (err error) {
 
 	return
 }
+
+func (r *Reader) FillObject(obj core.Object) error {
```

Comment:
This method is public, add comments for Go Doc.

Also add some tests for this method.